Valet/SynchronizableValet.swift

```diff
@@ -0,0 +1,259 @@
+//
+//  SynchronizableValet.swift
+//  Valet
+//
+//  Created by Dan Federman on 9/17/17.
+//  Copyright © 2017 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+
+@available(iOS 7.0.3, macOS 10.9, *)
+/// Reads and writes keychain elements that are synchronized with iCloud (supported on devices on iOS 7.0.3 and later).
+public final class SynchronizableValet: NSObject, KeychainQueryConvertible {
```

Comment:
devil's advocate - could we de-duplicate some of this?

Reply:
now that we've done this, subclass feels more valid. or.... base Valet but with a different flavor/accessibility